openpype/hosts/nuke/plugins/publish/collect_writes.py

```diff
@@ -69,11 +69,18 @@ def _set_existing_files_data(self, instance, colorspace):
             instance, collected_frames
         )
 
-        # inject colorspace data
-        self.set_representation_colorspace(
-            representation, instance.context,
-            colorspace=colorspace
-        )
+        set_colorspace = True
+
+        files = representation["files"]
+        if isinstance(files, list) and not files:
+            set_colorspace = False
+
+        if set_colorspace:
+            # inject colorspace data
+            self.set_representation_colorspace(
+                representation, instance.context,
+                colorspace=colorspace
+            )
```

Comment:
Looking over the code for how `files` is defined in the representation this could just be:
```suggestion
        # To avoid errors only try injecting colorspace 
        # if any existing files were found on disk
        if representation["files"]:
            # inject colorspace data
            self.set_representation_colorspace(
                representation, instance.context,
                colorspace=colorspace
            )
```